cmd/ig/daemon.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"path/filepath"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	gadgetservice "github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-service"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-service/api"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/runtime"
+)
+
+func addDaemonCommand(rootCmd *cobra.Command, runtime runtime.Runtime) {
+	daemonCmd := &cobra.Command{
+		Use:   "daemon",
+		Short: "runs Inspektor Gadget as a daemon",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return nil
+		},
+	}
+
+	var socket string
+	var gid int
+
+	daemonCmd.PersistentFlags().IntVarP(
+		&gid,
+		"daemon-group-id",
+		"",
+		0,
+		"Group id that the unix socket should use (if daemon-socket is set to e.g. unix:///path/to.socket)")
```

Comment:
Any thoughts between using the group id vs group name? Docker uses the name: `--group`. Perhaps the id is better because we don't have to worry about creating the group?

Reply:
We could rename it to `daemon-group` (or just `group` if you prefer that) and allow both; if it's numeric, use it as is, if it's a non-numeric, use `user.LookupGroupId()` to fetch the actual GID. WDYT?

Reply:
I think it's a good idea to support both, numeric and string. Just one concern, is it possible to name a group with a number?

Reply:
I already added support for both; a group name cannot be all-numeric according to [this post](https://unix.stackexchange.com/a/754021) (linking to [this](https://github.com/shadow-maint/shadow/blob/master/lib/chkname.c#L28)).

Reply:
> I think it's a good idea to support both, numeric and string.

Docker does that with a single flag. Here the group can be the group name or the number:
```
  -u, --user string                    Username or UID (format: <name|uid>[:<group|gid>])
```

> Just one concern, is it possible to name a group with a number?

It seems it depends on the distro, but that seems to be a bad idea:
https://unix.stackexchange.com/questions/287077/why-cant-linux-usernames-begin-with-numbers